src/state/stake/hooks.ts

```diff
@@ -61,13 +65,78 @@ export interface StakingInfo {
   readonly rewardTokens: Token[]
 }
 
+export const useMultiRewardPool = (): MultiRewardPool[] => {
+  const library = useProvider()
+  const farmSummaries = useFarmRegistry()
+
+  const [multiRewardPools, setMultiRewardPools] = React.useState<MultiRewardPool[]>([])
+
+  const call = React.useCallback(async () => {
+    const multiRwdPools: MultiRewardPool[] = []
+
+    await Promise.all(
+      farmSummaries.map(async (fs) => {
+        let poolContract = new Contract(fs.stakingAddress, DUAL_REWARDS_ABI, getProviderOrSigner(library) as any)
```

Comment:
I would use `MoolaStakingRewards__factory` here.

I would also use `useProvider` hook instead of `getProviderOrSigner`